File: Chapter13/VB/CommonTypes/WorkOrder.vb
<Serializable()> _
Public Class WorkOrder

  Private _model As String
  Public Property Model() As String
    Get
      Return _model
    End Get
    Set(ByVal Value As String)
      _model = Value
    End Set
  End Property

  Private _style As String
  Public Property Style() As String
    Get
      Return _style
    End Get
    Set(ByVal Value As String)
      _style = Value
    End Set
  End Property

  Private _color As String
  Public Property Color() As String
    Get
      Return _color
    End Get
    Set(ByVal Value As String)
      _color = Value
    End Set
  End Property

  Private _options As PricedItem()
  Public Property Options() As PricedItem()
    Get
      Return _options
    End Get
    Set(ByVal Value As PricedItem())
      _options = Value
    End Set
  End Property

  Private _id As String
  Public Property Id() As String
    Get
      Return _id
    End Get
    Set(ByVal Value As String)
      _id = Value
    End Set
  End Property

  ' public default constructor is required in serializable classes
  Public Sub New()
  End Sub

  Public Sub New(ByVal theModel As String, ByVal theStyle As String, _
                 ByVal theColor As System.Drawing.Color, _
                 ByVal theOptions As PricedItem())
    _model = theModel
    _style = theStyle
    _color = theColor.Name
    _options = theOptions

    _id = Guid.NewGuid().ToString()
  End Sub
End Class

File: Chapter13/C#/Common Types/WorkOrder.cs
using System;
using System.Drawing;
using System.Collections;

namespace Asap.Cars.CommonTypes
{
	/// <summary>
	/// Summary description for WorkOrder.
	/// </summary>
	[Serializable]
	public class WorkOrder
	{
    string model;
    public string Model
    {
      get {return model;}
      set {model = value;}
    }

    string style;
    public string Style
    {
      get {return style;}
      set {style = value;}
    }

    string color;
    public string Color
    {
      get {return color;}
      set {color = value;}
    }

    PricedItem[] options;
    public PricedItem[] Options
    {
      get {return options;}
      set {options = value;}
    }

    string id;
    public string Id
    {
      get {return id;}
      set {id = value;}
    }

    // public default constructor is required in serializable classes
    public WorkOrder()
    {
    }

    public WorkOrder(string theModel, string theStyle, Color theColor, PricedItem[] theOptions)
    {
      model = theModel;
      style = theStyle;
      color = theColor.Name;
      options = theOptions;

      id = Guid.NewGuid().ToString();
		}
	}
}
